// angularjs2-tabs/dist/index.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var core_1 = require("@angular/core");
var common_1 = require("@angular/common");
var tabs_component_1 = require("./src/tabs.component");
var tab_component_1 = require("./src/tab.component");
var tabs_directive_1 = require("./src/tabs.directive");
var tabs_pipe_1 = require("./src/tabs.pipe");
var tabs_service_1 = require("./src/tabs.service");
__export(require("./src/tabs.component"));
__export(require("./src/tabs.directive"));
__export(require("./src/tabs.pipe"));
__export(require("./src/tabs.service"));
var TabsModule = TabsModule_1 = (function () {
    function TabsModule() {
    }
    TabsModule.forRoot = function () {
        return {
            ngModule: TabsModule_1,
            providers: [tabs_service_1.TabsService]
        };
    };
    return TabsModule;
}());
TabsModule = TabsModule_1 = __decorate([
    core_1.NgModule({
        imports: [
            common_1.CommonModule
        ],
        declarations: [
            tabs_component_1.TabsComponent,
            tab_component_1.TabComponent,
            tabs_directive_1.TabsDirective,
            tabs_pipe_1.TabsPipe
        ],
        exports: [
            tabs_component_1.TabsComponent,
            tab_component_1.TabComponent,
            tabs_directive_1.TabsDirective,
            tabs_pipe_1.TabsPipe
        ]
    })
], TabsModule);
exports.TabsModule = TabsModule;
var TabsModule_1;
