// angularjs2-tabs/dist/src/tabs.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var core_1 = require("@angular/core");
var tab_component_1 = require("./tab.component");
// tabs.ts
var TabsComponent = (function () {
    function TabsComponent() {
    }
    TabsComponent.prototype.ngAfterContentInit = function () {
        // get all active tabs
        var activeTabs = this.tabs.filter(function (tab) { return tab.selected; });
        // if there is no active tab set, activate the first
        if (activeTabs.length === 0) {
            this.selectTab(this.tabs.first);
        }
    };
    TabsComponent.prototype.selectTab = function (tab) {
        this.tabs.map(function (_tab) {
            _tab.selected = false;
        });
        tab.selected = true;
    };
    return TabsComponent;
}());
__decorate([
    core_1.ContentChildren(tab_component_1.TabComponent),
    __metadata("design:type", core_1.QueryList)
], TabsComponent.prototype, "tabs", void 0);
__decorate([
    core_1.Input(),
    __metadata("design:type", String)
], TabsComponent.prototype, "classNames", void 0);
TabsComponent = __decorate([
    core_1.Component({
        selector: 'tabs',
        styles: ["\n        .ng2-tabs ul{\n            list-style: none;\n            padding: 0;\n            margin: 0;\n        }\n\n        .ng2-tabs ul:after {\n            content: \"\";\n            width: 100%;\n            height: 1px;\n            display: block;\n            clear: both;\n        }\n\n        .ng2-tabs ul li {\n            float: left;\n            margin-left: 2px;\n        }\n\n        .ng2-tabs ul li:first-child {\n            margin-left: 0;\n        }\n\n        .ng2-tabs ul li a {\n            display: block;\n            padding: 5px 10px;\n            text-decoration: none;\n            color: #000;\n            border: 1px solid #ccc;\n            border-bottom: none;\n            margin-bottom: -2px;\n            background-color: #ccc;\n        }\n\n        .ng2-tabs ul li.active a {\n            position: relative;\n            z-index: 2;\n            background-color: #fff;\n        }\n\n        .ng2-tabs .tab-content {\n            border: 1px solid #ccc;\n            padding: 10px;\n            box-sizing: border-box;\n        }\n    "],
        template: "\n        <div class=\"ng2-tabs\">\n            <ul>\n                <li *ngFor=\"let tab of tabs;  let i = index\" [class.active]=\"tab.selected\">\n                    <a href=\"javascript:void(0)\" (click)=\"selectTab(tab)\">{{tab.tabTitle}}</a>\n                </li>\n            </ul>\n            <div class=\"tab-content\">\n                <ng-content></ng-content>\n            </div>\n        <div>\n    "
    }),
    __metadata("design:paramtypes", [])
], TabsComponent);
exports.TabsComponent = TabsComponent;
